gdb fails to build.

On a '_almost vanilla_' Slackware 14.2 system:

**Focus:**<code><pre>gdb-dlfcn.o: In function `gdb_dlopen':
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/gdb/gdb-dlfcn.c:72: undefined reference to `dlopen'
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/gdb/gdb-dlfcn.c:80: undefined reference to `dlerror'
gdb-dlfcn.o: In function `gdb_dlclose':
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/gdb/gdb-dlfcn.c:113: undefined reference to `dlclose'
gdb-dlfcn.o: In function `gdb_dlsym':
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/gdb/gdb-dlfcn.c:103: undefined reference to `dlsym'
gdb-dlfcn.o: In function `gdb_dlclose':
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/gdb/gdb-dlfcn.c:113: undefined reference to `dlclose'
../bfd/libbfd.a(plugin.o): In function `try_load_plugin':
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/bfd/plugin.c:217: undefined reference to `dlopen'
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/bfd/plugin.c:224: undefined reference to `dlsym'
/mxe/tmp-gdb-i686-w64-mingw32.static/gdb-7.11.1/bfd/plugin.c:220: undefined reference to `dlerror'
collect2: error: ld returned 1 exit status
</pre></code>

Complete log here: https://gist.github.com/garry-iglesias/8b3c5cb7c2a2c74d72e51e86e7fc3717

Thanks.

--- a/src/dlfcn-win32.mk
+++ b/src/dlfcn-win32.mk
@@ -22,4 +22,13 @@ define $(PKG)_BUILD
     $(MAKE) -C '$(1)' -j '$(JOBS)'
     $(MAKE) -C '$(1)' -j 1 install
     $(MAKE) -C '$(1)' -j '$(JOBS)' test.exe testdll.dll
+
+    # create pkg-config file - mostly for psapi dependency
+    mkdir -p '$(PREFIX)/$(TARGET)/lib/pkgconfig'
+    (echo 'Name: $(PKG)'; \
+     echo 'Version: $($(PKG)_VERSION)'; \
+     echo 'Description: $(PKG)'; \
+     echo 'Libs: -ldl'; \
+     echo 'Libs.private: -lpsapi'; \
+    ) > '$(PREFIX)/$(TARGET)/lib/pkgconfig/dlfcn.pc'
 endef

--- a/src/gdb.mk
+++ b/src/gdb.mk
@@ -8,7 +8,7 @@ $(PKG)_SUBDIR   := gdb-$($(PKG)_VERSION)
 $(PKG)_FILE     := gdb-$($(PKG)_VERSION).tar.xz
 $(PKG)_URL      := http://ftp.gnu.org/pub/gnu/$(PKG)/$($(PKG)_FILE)
 $(PKG)_URL_2    := ftp://ftp.cs.tu-berlin.de/pub/gnu/$(PKG)/$($(PKG)_FILE)
-$(PKG)_DEPS     := gcc expat libiconv readline zlib
+$(PKG)_DEPS     := gcc dlfcn-win32 expat libiconv readline zlib
 
 define $(PKG)_UPDATE
     $(WGET) -q -O- 'http://ftp.gnu.org/gnu/gdb/?C=M;O=D' | \
@@ -21,6 +21,7 @@ define $(PKG)_BUILD
     cd '$(1)' && ./configure \
         $(MXE_CONFIGURE_OPTS) \
         --with-system-readline \
+        host_configargs="LIBS=\"`$(TARGET)-pkg-config --libs dlfcn`\"" \
         CONFIG_SHELL=$(SHELL)
     $(MAKE) -C '$(1)' -j '$(JOBS)'
 

